### PR TITLE
Add documentation for max_number_processors

### LINKED ISSUE #8148: [DOC]Add documentation for `cluster.ingest.max_number_processors` attribute
**What do you want to do?**
 Add the documentation for the newly added attribute `cluster.ingest.max_number_processors`
- [Y] Request a change to existing documentation
- [Y] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request.

***Version:** List the OpenSearch version to which this issue applies, e.g. 2.14, 2.12--2.14, or all.
 
**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by anandkrrai**
Closes #8148

### Description
_Describe what this change achieves._
Added documentation for `cluster.ingest.max_number_processors` attribute. 

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._
https://github.com/opensearch-project/documentation-website/issues/8148

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._
2.x

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ingest-pipelines/processors/index-processors.md [hunk 1]
```diff
@@ -69,6 +69,12 @@ Processor type | Description
 `urldecode` | Decodes a string from URL-encoded format.
 `user_agent` | Extracts details from the user agent sent by a browser to its web requests. 
 
+## Validations on Processors
```
Comment (dhwanilpatel) [inline _ingest-pipelines/processors/index-processors.md:72]: Can we call it as "Processor limit settings"?

File: _ingest-pipelines/processors/index-processors.md [hunk 2]
```diff
@@ -69,6 +69,12 @@ Processor type | Description
...
+
+We can configure the limits on the number of ingest processors that should be used. The limit can be configured using the attribute `cluster.ingest.max_number_processors`. The sum of the number of processors and the number of `on_failure` processors are considered for counting the total number of processors on which the limit would be applied.
```
Comment (dhwanilpatel) [inline _ingest-pipelines/processors/index-processors.md:74]: > The limit can be configured using the attribute

The limit can be configured using the cluster setting

File: _ingest-pipelines/processors/index-processors.md (hunk 1 above)
Comment (anandkrrai) [reply _ingest-pipelines/processors/index-processors.md]: Updated.

File: _ingest-pipelines/processors/index-processors.md (hunk 2 above)
Comment (anandkrrai) [reply _ingest-pipelines/processors/index-processors.md]: Updated

**Review by dhwanilpatel — APPROVED** [review]

**Review by vagimeli — APPROVED** [review]
Doc review completed

**Comment by vagimeli** [comment]
Applying "Done but waiting to merge" label to put this in the 2.17 editorial queue starting 9/11. This PR will be merged before the release.

File: _ingest-pipelines/processors/index-processors.md [hunk 3]
```diff
@@ -69,6 +69,12 @@ Processor type | Description
 `urldecode` | Decodes a string from URL-encoded format.
 `user_agent` | Extracts details from the user agent sent by a browser to its web requests. 
 
+## Processor limit settings
+
+Configure limits on the number of ingest processors using the cluster setting `cluster.ingest.max_number_processors`. The total number of processors includes both the number of processors and the number of [`on_failure`]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/) processors.
```
Comment (natebower) [inline _ingest-pipelines/processors/index-processors.md:74]:
```suggestion
You can limit the number of ingest processors using the cluster setting `cluster.ingest.max_number_processors`. The total number of processors includes both the number of processors and the number of [`on_failure`]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/) processors.
```

File: _ingest-pipelines/processors/index-processors.md [hunk 4]
```diff
@@ -69,6 +69,12 @@ Processor type | Description
...
+
+The default value for `cluster.ingest.max_number_processors` is `Integer.MAX_VALUE`. Adding more processors than the configured value in `cluster.ingest.max_number_processors` will throw an `IllegalStateException`.
```
Comment (natebower) [inline _ingest-pipelines/processors/index-processors.md:76]:
```suggestion
The default value for `cluster.ingest.max_number_processors` is `Integer.MAX_VALUE`. Adding a higher number of processors than the value configured in `cluster.ingest.max_number_processors` will throw an `IllegalStateException`.
```

**Review by natebower — COMMENTED** [review]
@anandkrrai @vagimeli Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**